catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/ingest/PostIngestConsumer.java

```diff
@@ -0,0 +1,126 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.camel.component.catalog.ingest;
+
+import java.util.Hashtable;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.impl.DefaultConsumer;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.camel.component.catalog.CatalogEndpoint;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.Response;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+
+/**
+ * Camel Consumer Post Ingest plugin used to provide the responses passed to Post Ingest Plugins
+ * to camel routes.
+ */
+public class PostIngestConsumer extends DefaultConsumer implements PostIngestPlugin {
+    public static final String CREATE = "create";
+
+    public static final String UPDATE = "update";
+
+    public static final String DELETE = "delete";
+
+    public static final String ACTION = "action";
+
+    private static final int NUM_THREADS = 5;
+
+    private CatalogEndpoint endpoint;
+
+    private ServiceRegistration registration;
+
+    private BlockingQueue<Runnable> blockingQueue = new ArrayBlockingQueue<>(NUM_THREADS);
+
+    private ExecutorService threadExecutor = new ThreadPoolExecutor(NUM_THREADS,
+            NUM_THREADS,
+            0L,
+            TimeUnit.MILLISECONDS,
+            blockingQueue);
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostIngestConsumer.class);
+
+    public PostIngestConsumer(CatalogEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        processRequest(input, CREATE);
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        processRequest(input, UPDATE);
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        processRequest(input, DELETE);
+        return input;
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        super.doStop();
+        if (registration != null) {
+            registration.unregister();
+        }
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        Hashtable<String, String> props = new Hashtable<String, String>();
```

Comment:
Updated to use the Dictionary abstract class that the registerService defines in its method signature. Updated to use `<>` on the right hand side